### DISCUSSION TITLE
Wemod on Heroic

### DISCUSSION THREAD

**Discussion by mzigaib** [DeckCheatz/wemod-launcher#116] [Q&A] [answered]
I could make it work on steam but how can I make it work on Heroic, I tried to just add the line on the "arguments" slot but nothing happened, is it possible to make it work there?

**Comment by marvin1099** [comment]
i was using it in lutris a lot,
there it works, so it probably works on heroic two.
Here is what i need you to do:
1. Delete the wemod.log file if it exists.
2. Close heroic fully.
3. Open the terminal.
4. Type `heroic` or the full path to the heroic launcher (for example the appimage filepath would work).
5. If you are using heroic over flatpak you probaly have less sucess sine flatpak is not fully supported.
6. But if you want to try it you will ned to run `flatpak run com.heroicgameslauncher.hgl`. 
7. Then start the game you want with wemod.
8. Copy the terminal output into here after you have closed the game or the after the game has crashed with wemod.
9. Also upload the wemod.log if it exists now.

**Comment by marvin1099** [comment]
Oh you will also need to, follow https://github.com/DaniAsh551/wemod-launcher/wiki/Full-guide#step-35-set-settings-of-steam-flatpak-if-in-use if you use flatpak, the steam stuff will need to be relaced by heroic paths of course

**Comment by mzigaib** [comment]
Thanks for the feedback!
I will try that and let you know if it worked.

**Reply by marvin1099** [reply]
If possible use the appimage or the store package for heroic, 
like I mentioned flatpak support is really not that good, 
The reason why its not that good, is that I don't really known how flatpak works.
So its not going to change unless a new maintainer comes to the project.

**Comment by marvin1099** [comment]
Here is a wiki entry for [Using External Launchers](https://github.com/DaniAsh551/wemod-launcher/wiki/Using-External-Launchers),
but to make it short any launcher
that supports custom startup commands
can be used with the wemod-launcher.
To make any launcher other then steam work,
all that you will need is modification to the startup command,
in the same way that the steam command is modified.
Of course as i mentioned its also important that the external launcher 
is not in a sandbox like for flatpak or in a snap (snaps don't work at all, pretty sure).
Other then that i think this issue can be closed.
If you still have questions about this just ask.

**Comment by mzigaib** [comment]
I was just hoping that someone could help me in pointing where I could do this specifically on heroic, I already tried the commands slot but no success

**Reply by marvin1099** [reply]
Not shure how heroic works, but you can do it with lutris as well, I will see how heroic works, since you sind say anything I thought that you figured it out.

**Reply by mzigaib** [reply]
Not yet, but thanks for the help

**Comment by marvin1099** [comment, marked as answer]
So i see, heroic wont support a command prefix, like lutris is doing.
To set up wemod over heroic you will need to manually install wemod.
To do that follow this [new guide](https://github.com/DaniAsh551/wemod-launcher/wiki/Using-the-Heroic-Launcher) i have added, it should make it work.
But i can also recommend [Lutris, if you like to try that](https://github.com/DaniAsh551/wemod-launcher/wiki/Using-External-Launchers).
Keep on mind as before you will still need the appimage version for heroic
or one from you distro repos, so not the flatpac version.

**Reply by mzigaib** [reply]
Oh Ok, I will take a look and I will tell if I can make this work.
And I can´t thank you enough for the trouble doing the guide man, I really appreciate it.

**Reply by mzigaib** [reply]
I could make it work on Lutris which is great, I can work with that yay!
There are some games the need to be run inside the epic launcher like red dead redemption 2, so in this case is it possible to make wemod to work in this situation?

**Reply by marvin1099** [reply]
As I have mentioned follow the [new heroic guide](https://github.com/DaniAsh551/wemod-launcher/wiki/Using-the-Heroic-Launcher), that is the explaination for making heroic work.

**Reply by mzigaib** [reply]
Ok I will try that, thanks!

**Reply by marvin1099** [reply]
I will close this for now, if any problems appear, don't hesitate to ask

**Comment by RegressiaUK** [comment]
@marvin1099 I am not sure what's happening or if this is just very out of data but now when you follow this https://github.com/DeckCheatz/wemod-launcher/wiki/Using-the-Heroic-Launcher when it try loading it 

```bash
WeModLog path was not given or invalid using path 'wemod.log'
If you don't want to generate a log file, use WEMOD_LOG='' or set the config to WeModLog=''
Adding script name to config
Adding script version to config
The script wemod-launcher is running on version 1.532
Package 'FreeSimpleGUI' is missing
Pip error occurred:
	b''
	b'/bin/sh: 1: pip: not found\n'
Pip error occurred:
	b''
	b'/usr/bin/python3: No module named pip\n'
Pip not found. Downloading...
Pip finished and returned:
	b'Defaulting to user installation because normal site-packages is not writeable\nCollecting FreeSimpleGUI (from -r /home/susy/wemod-launcher/requirements.txt (line 1))\n  Downloading freesimplegui-5.2.0.post1-py3-none-any.whl.metadata (45 kB)\nRequirement already satisfied: requests in /usr/lib/python3/dist-packages (from -r /home/susy/wemod-launcher/requirements.txt (line 2)) (2.25.1)\nDownloading freesimplegui-5.2.0.post1-py3-none-any.whl (733 kB)\n   \xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81\xe2\x94\x81 733.6/733.6 kB 11.0 MB/s eta 0:00:00\nInstalling collected packages: FreeSimpleGUI\nSuccessfully installed FreeSimpleGUI-5.2.0.post1\n'
	b"  WARNING: The scripts fsghelp, fsgissue, fsgmain, fsgsettings and fsgver are installed in '/home/susy/.local/bin' which is not on PATH.\n  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.\n"
No updates available
Error: The WINE environment variable needs to be set if using external runners. Exiting...
Error: wine not found.
The WINE environment variable needs to be set if using external runners. Exiting...



The script wemod-launcher is running on version 1.532
No updates available
Error: The WINE environment variable needs to be set if using external runners. Exiting...
```
![image](https://github.com/user-attachments/assets/b61c3b2a-f915-4964-a6c2-d6e199fd6ce8)
![image](https://github.com/user-attachments/assets/6130a82b-68d1-46ff-8a91-3c73f6e30ccb)
![image](https://github.com/user-attachments/assets/e722d687-3b4f-4d7e-8732-1ecb9dbffde0)

As you can see all the settings work and I really do have GE as the game works fine.

**Reply by marvin1099** [reply]
You will need to run this in the terminal:
```
WINEPREFIX=$YOUR-WINE-PREFIX-FOLDER-HERE WINE=$(whereis wine | awk '{print $2}') /home/$USER/Apps/Games/wemod-launcher/wemod $(whereis wine | awk '{print $2}') /home/$USER/Apps/Games/wemod-launcher/wemod.bat
```
You removed " WINE=" in the command.

Of course still replace the folder locations.
But be careful this time not to remove anything additional.

**Reply by RegressiaUK** [reply]
Are I see, is it recommeneded to download or build for the prefix, and yep that was my mistake there even copying it 3 times a redoing it I seem to of did that each time.

**Comment by RegressiaUK** [comment]
Or after it's started to install and everything and set things up I got this 
![image](https://github.com/user-attachments/assets/e3b0cce9-d9c5-445a-ad7e-92e0446bb8a6)

**Reply by marvin1099** [reply]
Please read the main README.
But here a refresher.
> 9. Select Yes/Ok until no more windows appear.  
       All rundll32.exe errors can safely be ignored (by clicking no).  
       WeMod should start with the game.

**Comment by RegressiaUK** [comment]
OK I am so confused when I did the download version it messed up and made my game no detect it's self anymore GTA in this case so I had to redo it and set it to build, even though it says it's not the best option, but it's the only way for the game to know where really is. but I installed 48 to GE-Proton but wemod is telling me to swing it saying it's not installed 

![image](https://github.com/user-attachments/assets/8b74daea-20c8-4b7b-bbe8-2c0a17ce36ec)


Your guys are doing good though I not been this close in a long time, any tips to get it past the line.

**Reply by RegressiaUK** [reply]
I have installed 48 and even set it to win11 but still seems to say it can't find the right .NET so close but no cigar

**Reply by marvin1099** [reply]
This was not fully sucssesfull.
```
WINEPREFIX=$YOUR-WINE-PREFIX-FOLDER-HERE WINE=$(whereis wine | awk '{print $2}') /home/$USER/Apps/Games/wemod-launcher/wemod $(whereis wine | awk '{print $2}') /home/$USER/Apps/Games/wemod-launcher/wemod.bat
```
Did you click download when ased to get a prefix.
This is needed when this error appears.
You do not want to copy the prefix (select no when asked).
Then it should work.

To restart the install remove the prefix.
Then retry the command above.
Still replace the prefix path to the correct one.

**Reply by RegressiaUK** [reply]
What do you mean copy that option was never given I had download and build options only and I just told you if I use the download version rockstar no longer finds my game it brakes as it's no longer in the prefix the game is loading from I will give it another go what's the best way to unistall just delete the folder?

![image](https://github.com/user-attachments/assets/5332be8a-a085-432f-9181-e638793c28d6)

**Reply by RegressiaUK** [reply]
![image](https://github.com/user-attachments/assets/f895fb6d-d210-427b-8dc9-1e3b9e7e5fbd)

I got this but when i was posting it here it conteinued on it's own so not sure what option it selected

**Reply by RegressiaUK** [reply]
Now it's completed it did this lol 

![image](https://github.com/user-attachments/assets/75cab088-f824-44fe-813e-cb34245c5988)

**Reply by RegressiaUK** [reply]
In the error logs its reporting windows 7 but wemod/.net wont install on 7 I am not sure if it's the downloaded profile or the game profiles whats set it to 7 and this is default for GE-Proton if this is the case

**Reply by marvin1099** [reply]
You seem to be on ProtonGE 6.36
![image](https://github.com/user-attachments/assets/f895fb6d-d210-427b-8dc9-1e3b9e7e5fbd)
This is very outdated.
The newest version should be ProtonGE10.4
Try to grab the Proton Version via ProtonUp.

Or maby in the heroic settings you can do it.
Not shure.
But you need ProtonGE8.1 or later.

**Reply by RegressiaUK** [reply]
No I am on Proton GE10.4 but the problem is when it's loaded via the downloaded option my game wont load rockstar lanuncher no longer is able to find my game says install when it's already there.

![image](https://github.com/user-attachments/assets/045cce43-20af-469d-a29b-017126ae0c6f)

![image](https://github.com/user-attachments/assets/b7fe7e8a-6fcc-4d96-b509-e72c18277302)

**Reply by RegressiaUK** [reply]
Removing the bat option the game will load as it's meant to the profile downloads and the one the game works from don't seem to link there for the game no longer seen.

**Reply by marvin1099** [reply]
Alls this is doing is installing the wemod launcher:
```
WINEPREFIX=$YOUR-WINE-PREFIX-FOLDER-HERE WINE=$(whereis wine | awk '{print $2}') /home/$USER/Apps/Games/wemod-launcher/wemod $(whereis wine | awk '{print $2}') /home/$USER/Apps/Games/wemod-launcher/wemod.bat
```

It won't start the game.
That is normal.

You just have to continue to follow the guide.
I think the wine version is also the problem.
Its to old.
It should also be something like 10.0
Make shure you system is up to date.
Check you wine version via
```
wine --version
```
If its below 8.0 thats also a problem.
You will have to download a newer wine version manualy then set WINE="bla"
To the newer wine path.

Not shure if I fully understand your isse.
If you are not a native speaker you may want to use deepl.com to translate your issue.

**Reply by RegressiaUK** [reply]
When I go to run in Wemod I mean and you select the z path for the game to load it, it dose not detect the game anymore.

I read what I have put and sorry but it reads fine, I am English 100% little dyslexic so my wording can be a little messed up but sorry no way in hells chance does it not read a native speaking that wasn't needed to be included at all but anyway.

I just updated wine pushing the stable build and now it's stating 10

Do I need to do the install again or just retry it?

**Reply by marvin1099** [reply]
I just asked you about the native speaker thing because I was not shure if I understood you correctly (not required of course).

Delete the prefix folder and rerun the command.

Then continue with the guide as usual.

Maby send some screenshots for that:
> When I go to run in Wemod I mean and you select the z path for the game to load it, it dose not detect the game anymore.

**Reply by RegressiaUK** [reply]
![image](https://github.com/user-attachments/assets/fb955590-349b-46f6-95d2-d54c9770d24c)

This is what's happening its acting like this a different profile, the game is no longer finding it's files and cashe there for no longer working and no longer will show the game is installed when I have logged in.

**Reply by RegressiaUK** [reply]
I will redo the prefix folder how do I get the delete folder dialog window up ? or will just deleting the hole wemod folder do it and recreate it?

**Reply by RegressiaUK** [reply]
So after first installing it it does this 

![image](https://github.com/user-attachments/assets/fa3fd7f2-ce71-40e5-b1b8-39df43fb11f0)

This is not yet loaded in to game launhcer this is the installer what did this. The buttons are the same dotnet ones.

**Reply by marvin1099** [reply]
> So after first installing it it does this
> 
> ![image](https://private-user-images.githubusercontent.com/90859294/456552766-fa3fd7f2-ce71-40e5-b1b8-39df43fb11f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTAyNTYxODYsIm5iZiI6MTc1MDI1NTg4NiwicGF0aCI6Ii85MDg1OTI5NC80NTY1NTI3NjYtZmEzZmQ3ZjItY2U3MS00MGU1LWIxYjgtMzlkZjQzZmIxMWYwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjE4VDE0MTEyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2YzMwOGExYWM2YTE1NmY1YWY5OGM2OGQ5OTU5YThlZDcyOWE3NjcxMTNmZGZiMzI3ZDZmODRmOGNhOTNlZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.W1SDjMAyyCVexuwM75UpLf0H5tlaPK2qUAnjCPCFFsk)
> 
> This is not yet loaded in to game launhcer this is the installer what did this.

This is expected.

**Reply by RegressiaUK** [reply]
So when I open it with the game lanucher Wemod opens asks me to login, all this I was able to do again so dotnet is working in the games prefix but the game is not detected 

So I search gta give it the z path in wemod and then it's all broken from there 
![image](https://github.com/user-attachments/assets/d179d871-ca76-43c0-9236-ca73cba4cbaf)
I am already logged in, I am already playing the game and have a  save but it seems the wemod app isn't running on the same instence as my game there for they are not fully taking to one another and failing to let the game work as soon as I am in the lanucher it asks me to install the game.

**Reply by RegressiaUK** [reply]
See asking me to install 
![image](https://github.com/user-attachments/assets/4c7a9536-fef9-40a7-b888-5e02c495f5f6)

**Reply by RegressiaUK** [reply]
![image](https://github.com/user-attachments/assets/8b366e30-1262-4311-929d-2d15f5f33fa8)

**Reply by RegressiaUK** [reply]
But if I remove /home/susy/wemod-launcher/wemod.bat the game will load as it's meant to but after I run the installer and it adds it to the prefix it seems to also delete my game save folder so I have to re bloody do it I had to re do the opening game now 3 times.


I even tried the run exe on prefix dose the same thing acts like the files and folders it runs from are not the same as the prefix so brakes the game

**Reply by marvin1099** [reply]
Ok.
Can you  set the working dirrectory?
I think heroic may change the working dirrectory to be at wemod.bat
The working dirrectory needs to be set to where your exe is.
Try that.
like if the game is at "/test/hi.exe" the working dir needs to be at "/test/"

**Reply by marvin1099** [reply]
> Ok. Can you set the working dirrectory? I think heroic may change the working dirrectory to be at wemod.bat The working dirrectory needs to be set to where your exe is. Try that. like if the game is at "/test/hi.exe" the working dir needs to be at "/test/"

@TheShadowModsUK if this wont work you can also try to copy the bat to the same place as the game.
And then use the new path in the "select an exe to run" box.

**Reply by RegressiaUK** [reply]
Wine Tab:
![image](https://github.com/user-attachments/assets/87f612e0-968a-4f8b-b0ba-d0d8b087bab9)
![image](https://github.com/user-attachments/assets/c55fce16-a53b-4dd6-8d1a-83e68ecc506a)

Other Tab:
![image](https://github.com/user-attachments/assets/11dee949-ff12-4f7e-9b54-7541bb757d55)

Advanced Tab:
![image](https://github.com/user-attachments/assets/0aa9bb08-9c70-4dda-8f9c-9b615efdddbb)
![image](https://github.com/user-attachments/assets/e10868e9-82f5-4494-999c-1d394f7493b6)


But you would of thought the Run EXE on prefix would override all folder changes and that also did the same thing.

want me to try or do anything else just ask.

Also thank you for helping as much as you have so far, you helped at first what was human error, ya i found more problems lol but your still trying so thanks.

**Reply by RegressiaUK** [reply]
> > Ok. Can you set the working dirrectory? I think heroic may change the working dirrectory to be at wemod.bat The working dirrectory needs to be set to where your exe is. Try that. like if the game is at "/test/hi.exe" the working dir needs to be at "/test/"
> 
> @TheShadowModsUK if this wont work you can also try to copy the bat to the same place as the game. And then use the new path in the "select an exe to run" box.

Just the wemod.bat file or do you want me to move other files as well?

**Reply by marvin1099** [reply]
Actually Link the folder wemod_bin to the game folder as well.
Then it should work.

**Reply by marvin1099** [reply]
If you use kde, trying to move a folder will give you a link option.
Otherwise right click in game folder (empty space), then new, then link, then paste /home/susy/wemod-launcher/wemod_bin
as path, set the name as wemod_bin

**Reply by RegressiaUK** [reply]
I am using popOS what is Ubuntu/Gnome based and I don't have any option to link a file let along a folder know of any other way we could do it.

**Reply by RegressiaUK** [reply]
Are you asking me to set up a symbolic link ?and if so shouldn't it be of the hole wemod-launcher folder so it can access the bat and data folder

**Reply by RegressiaUK** [reply]
Are bingo I have create link now and the folder created in the right place, I had to renable a setting in Gnome File

**Reply by RegressiaUK** [reply]
Sadly now it just says loading/playing and nothing opens

**Reply by RegressiaUK** [reply]
I just also pointed it to the wemod exe and same as if you ran it via the dot bat file before lanucher don't find the game soon as you logged in to the rockstar lanucher, I m sure Heroic is doing something to tell the exe where it is so it knows it's in that folder as the lanucher looks for C:\program files\rockstar games or something I don't know I am out of ideas lol

**Reply by marvin1099** [reply]
> Are you asking me to set up a symbolic link ?and if so shouldn't it be of the hole wemod-launcher folder so it can access the bat and data folder

The bat and the wemod_bin should be in the same folder.
If you link the launcher it will be in a subfolder.
Yes "syslink = link" basically this is what i mean.

In your case you probably want to use the Rockstar / GTA launcher directory actually, to link the files to.
That way the Rockstar / GTA launcher should start with wemod and should find GTA.

**Reply by marvin1099** [reply]
![image](https://github.com/user-attachments/assets/52adb659-0833-467f-abc4-0d8b96ca0373)
Like this (this is a mookup of course; the launcher may just be named rockstar-launcher or something like that).

**Reply by RegressiaUK** [reply]
That's pretty look UI lol I have it like so
![image](https://github.com/user-attachments/assets/4e232573-d4aa-455e-a4b3-1ed15dc19118)

So I will change that in a moment

**Reply by marvin1099** [reply]
@TheShadowModsUK Actually the mookup is wrong i linked the "wemod_data" folder but i should have linked the "wemod_bin" folder

**Reply by RegressiaUK** [reply]
> @TheShadowModsUK Actually the mookup is wrong i linked the "wemod_data" folder but i should have linked the "wemod_bin" folder

I will make sure its bin then :P

**Reply by RegressiaUK** [reply]
Bat file opened closed then opened wemod then it did the same problem the lanucher isn't finding the game I am pretty much stumbed at this point, if this was a windows box we could just catch the exe and it checks it if it's running and off we go but this is just a piss take lol

I am unsure if we will get this working I believe the problem is Rockstars lanucher hard coding it's paths or saving the paths else where to run the game but making the game need the lanucher to boot

**Reply by marvin1099** [reply]
Well there is still a non-convenient option.
Reset the start options in heroic.
Start the game as usual (with heroic).
Then use this button to start wemod (in the new window select the wemod.exe in the linked wemod_bin folder),  
while The rockstar launcher allready runns.
![image](https://github.com/user-attachments/assets/e722d687-3b4f-4d7e-8732-1ecb9dbffde0)
its the button way on the bottom right (cut off).
It says something like: run an exe in the prefix.

**Reply by RegressiaUK** [reply]
One thing I did just try was link the folders to where the lanucher wanted the files what worked to load the game but for some reason it just closed wemod so ya I will try that option see if that works.

**Reply by RegressiaUK** [reply]
There is some kind of hold over the prefix where it will only fully load on exe at a time when you try it wont load wemod past  black screen and if you try doing the run first of wemod and then play the game the game wont load.

I think I am seeing the same problem when I loading it via wemod after the files have been pointed  via link it's just not enable to handle two apps with full UI's basic bat files seems ok but after that it gets upet.

**Reply by marvin1099** [reply]
You could try to set the game exe to simply: cmd.exe
Then you can start any app via that prefix by typing the full path in the cmd and hitting enter.
Best to use: start "" "FULLPATH"
And see if you can make it work that way.
First for FULLPATH paste the GTA-Laucher path.
See it it works like that.
Here you can also use: cd "PATH-WHERE-GTA-LAUNCHER-IS" 
After that you retry to run the GTA LAUNCHER,
with: start "" "FULLPATH"
That may work.

If you cant get this far i dont think i can do a lot.

And then if it works run in the cmd:
start "" "FULLPATHWEMOD"
FULLPATHWEMOD replace with the linked wemod exe.


You can still try with WINE instead of proton.
While Proton is better for games wine may just work as well.
If you like to try it you need to select Wine version system or something like that.

You may also want to consider using lutris.
It may be a problem with Heroic.
Not shure.
For that you would have to add a new game in lutris set the wine prefix to heroics prefix and set the gta launcher path.
Then you can also use the other guide.
Its a bit nicer the Heroic solution is a bit of a workaround you know.

**Reply by RegressiaUK** [reply]
I think I will give this a brake today, you have been a star, I will try this again in a day or two, I was also helping on anther project doing so debuging and was spat on by the dev's so I little f'd right now so the game still works at lest, ya I can't use wemod, this was for the wife so she could use it she's a light gamer and just loves to kill everything in GTA lol

I was going to do the mod way but sadly the scripthook also needs a update lol
